application/clicommands/ListCommand.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Icinga\Module\Reporting\Clicommands;
+
+use Icinga\Module\Reporting\Cli\Command;
+use InvalidArgumentException;
+use ipl\Sql\Select;
+
+class ListCommand extends Command
+{
+    /**
+     * List reports
+     *
+     *  USAGE:
+     *
+     *   icingacli reporting list --sort id|name|author --direction ASC|DESC (default DESC)
+     */
+
+    public function indexAction()
+    {
+        $sort = $this->params->get('sort', 'r.mtime');
```

Comment:
I was a bit hasty. You still need to manually validate the sort parameter so that it only accepts either `id, name, author`. For all the others an exception has to be thrown.

```php
new InvalidArgumentException(sprintf('Sorting by %s is not supported', $sort));
```